src/main/java/io/jenkins/plugins/opentelemetry/init/OtelJulHandler.java

```diff
@@ -51,51 +69,59 @@ public class OtelJulHandler extends Handler implements OtelComponent {
      */
     @Override
     public void publish(LogRecord logRecord) {
-        LogRecordBuilder logBuilder =  otelLogger.logRecordBuilder();
-        // message
-        String message = FORMATTER.formatMessage(logRecord);
-        if (message != null) {
-            logBuilder = logBuilder.setBody(message);
+        if (disabled) {
+            return;
         }
+        try {
+            LogRecordBuilder logBuilder = otelLogger.logRecordBuilder();
+            // message
+            String message = FORMATTER.formatMessage(logRecord);
+            if (message != null) {
+                logBuilder = logBuilder.setBody(message);
+            }
+
+            // time
+            Instant timestamp = logRecord.getInstant();
+            logBuilder = logBuilder.setEpoch(timestamp);
+
+            // level
+            Level level = logRecord.getLevel();
+            if (level != null) {
+                logBuilder = logBuilder
+                    .setSeverity(levelToSeverity(level))
+                    .setSeverityText(logRecord.getLevel().getName());
+            }
+
+            AttributesBuilder attributes = Attributes.builder();
+
+            // throwable
+            Throwable throwable = logRecord.getThrown();
+            if (throwable != null) {
+                attributes.put(SemanticAttributes.EXCEPTION_TYPE, throwable.getClass().getName());
+                attributes.put(SemanticAttributes.EXCEPTION_MESSAGE, throwable.getMessage());
+                StringWriter writer = new StringWriter();
+                throwable.printStackTrace(new PrintWriter(writer));
+                attributes.put(SemanticAttributes.EXCEPTION_STACKTRACE, writer.toString());
+            }
+
+            if (captureExperimentalAttributes) {
+                Thread currentThread = Thread.currentThread();
+                attributes.put(SemanticAttributes.THREAD_NAME, currentThread.getName());
+                attributes.put(SemanticAttributes.THREAD_ID, currentThread.getId());
+            } else {
+                attributes.put(SemanticAttributes.THREAD_ID, logRecord.getThreadID());
+            }
 
-        // time
-        // TODO (trask) use getInstant() for more precision on Java 9
-        long timestamp = logRecord.getMillis();
-        logBuilder = logBuilder.setEpoch(timestamp, TimeUnit.MILLISECONDS);
-
-        // level
-        Level level = logRecord.getLevel();
-        if (level != null) {
             logBuilder = logBuilder
-                .setSeverity(levelToSeverity(level))
-                .setSeverityText(logRecord.getLevel().getName());
+                .setAllAttributes(attributes.build())
+                .setContext(Context.current());// span context
+
+            logBuilder.emit();
+        } catch (RuntimeException e) {
+            System.err.println("Exception sending logs to OTLP endpoint, disable OTelJulHandler");
```

Comment:
Is there a reason not to use the logger?

Reply:
Yes because the `OTelJulHandler` is registered as a `Handler` of the logger so it's a kind of infinite loop

Reply:
That was my guess; we can put a comment for the future. It happened to me that people as for something similar in the SSH Build Agents plugin and people made some PRS changing it to use a logger. A simple comment avoids those PRs and have to explain several times that it is not possible to use a logger.

Reply:
```suggestion
            // directly use `System.err` rather than the `logger` 
            //because the OTelJulHandler is a handler of this logger, risking an infinite loop
            System.err.println("Exception sending logs to OTLP endpoint, disable OTelJulHandler");
```